selfhost/typechecker.jakt

```diff
@@ -7492,9 +7494,9 @@ struct Typechecker {
         Garbage => (None, expr)
     }
 
-    fn typecheck_var_decl(mut this, var: ParsedVarDecl, init: ParsedExpression, scope_id: ScopeId, safety_mode: SafetyMode, span: Span) throws -> CheckedStatement {
+    fn typecheck_var_decl_without_adding(mut this, var: ParsedVarDecl, init: ParsedExpression, scope_id: ScopeId, safety_mode: SafetyMode, span: Span) throws -> (CheckedVariable, CheckedExpression)? {
```

Comment:
nit: adding what?

Reply:
Without... registering the variable in a scope. I'll rename the other version to `typecheck_and_register_var_decl`, or inline it if it's only used in one place.

Reply:
Sure, any names works so long as it's less ambiguous